core/rpc/src/author/mod.rs

```diff
@@ -162,42 +161,45 @@ impl<B, E, P, RA> AuthorApi<ExHash<P>, BlockHash<P>> for Author<B, E, P, RA> whe
 			let best_block_hash = self.client.info().chain.best_hash;
 			let dxt = <<P as PoolChainApi>::Block as traits::Block>::Extrinsic::decode(&mut &xt[..])
 				.map_err(error::Error::from)?;
-			Ok(self.pool
-				.submit_and_watch(&generic::BlockId::hash(best_block_hash), dxt)
-				.boxed()
-				.compat()
-				.map_err(|e| e.into_pool_error()
-					.map(error::Error::from)
-					.unwrap_or_else(|e| error::Error::Verification(Box::new(e)).into())
-				))
+			Ok(
+				self.pool
+					.submit_and_watch(&generic::BlockId::hash(best_block_hash), dxt)
+					.map(|res| res.map_err(|e|
+						e.into_pool_error()
+							.map(error::Error::from)
+							.unwrap_or_else(|e| error::Error::Verification(Box::new(e)).into())
```

Comment:
Why don't we use a second `map_err` here?
We could also just make one `map_err` that outputs the correct error directly.